joblib/_parallel_backends.py

```diff
@@ -553,20 +586,22 @@ def effective_n_jobs(self, n_jobs):
 
     def apply_async(self, func, callback=None):
         """Schedule a func to be run"""
-        future = self._workers.submit(SafeFunction(func))
-        future.get = functools.partial(self.wrap_future_result, future)
+        future = self._workers.submit(func)
         if callback is not None:
             future.add_done_callback(callback)
         return future
```

Comment:
The result is now fetch in `fetch_result_callback` that calls the `result` so we do not need to warp it anymore.

the `DelayedResult` is only used for the sequential backend. Maybe a better name would be `SequentialResult` to make this explicit.

Reply:
> The result is now fetch in fetch_result_callback that calls the result so we do not need to warp it anymore.

Ok.

> the DelayedResult is only used for the sequential backend. Maybe a better name would be SequentialResult to make this explicit.

+1. I would like to get rid of the sequential backend completely at some point but probably for another PR.

Reply:
:+1:

Once we get the effective `n_jobs=1`, we should simply special case and return a generator that calls the function directly. The would actually simplify this PR, maybe we could do it directly in it? This would avoid introducing this object at all.